### DISCUSSION TITLE
The first distributed miningpool running in the Filecoin Mainnet

### DISCUSSION THREAD

**Discussion by Joss-Hua** [filecoin-project/community#164] [Mining]
*After half a year, Venus (go-filecoin) has been running the first distributed mining pool in the Filecoin Mainnet, from the stagnation of development.*

Now, the latest version of Venus has been released on GitHub. If you need any support during mining, please feel free to contact Venus team.

- Slack ([fil-venus channel of filecoinproject.slack.com](https://filecoinproject.slack.com/archives/CEHHJNJS3))
- GitHub (https://github.com/filecoin-project/venus/discussions)
- WeChat (Filecoin-Venus)
- E-mail (venus@ipfsforce.com)

**Venus node: f0128788**

In January this year, Venus v0.9.0 was released, and the first Venus node f0128788 based on this version was launched on the Filecoin Mainnet. After four months of continuous observation and upgrading, through the adjustment of different sealing speeds, mining strategies and configurations, f0128788's indicators, such as rate of isolated block and gas of overestimation burn, are all in the normal range. At present, the Quality Adjusted Power of this node is close to 1PiB, and its Lucky Value of the past 30 days is 107%.

![image](https://user-images.githubusercontent.com/75723122/118358368-0acd5600-b5b1-11eb-9e07-a29757b53896.png)

**Distributed mining pool connected to several nodes**

The Venus technical architecture is designed to include venus, venus-miner, venus-sealer, venus-messenger, venus-wallet, venus-auth, and also the upcoming venus-market. In the classic architecture, there are venus, venus-miner, venus-auth, venus-messenger and venus-market as shared components, besides venus-sealer and venus-wallet as local independent components.
 
Shared components can be provided by miners, miners unions or third-party services. Miners only need to focus on the operations and maintenance of local independent components. If miners deploy all or part of the shared components locally, they can deploy distributed mining pool more flexibly.

![image](https://user-images.githubusercontent.com/75723122/118358310-c8a41480-b5b0-11eb-89dd-d6b7172b7c69.png)
(Venus architecture diagram)
 
The Venus components has passed the tests of 2K, calibration, Nerpa and Mainnet network. Based on f0128788, we built a complete set of Venus components and the first distributed mining pool in the Filecoin Mainnet. This mining pool has been connected to several distributed nodes, which means that Filecoin has had the first real distributed mining pool.

![image](https://user-images.githubusercontent.com/75723122/118358389-1fa9e980-b5b1-11eb-8282-af2996a104b5.png)
(The miners)
 
Venus distributed mining pool is more suitable for small miners, rather than big miners. In the first step, we try to connect more nodes below 2PiB to join Venus mining pool, or guide miners to deploy their own distributed mining pool. The implementation of the current version can save hardware resources, ensure the stability of block rewards, enhance security of wallet address, and improve the speed of sending message on chain, for the miners inside the mining pool.

**Free service, welcome to access**

At present, the Filecoin Network storage power has reached 5EiB+, and the block rewards of small mining clusters has declined. Due to the rapid development of Filecoin ecosystem, the demand for hardware is gradually increasing, and high cost has become one of the pain points. Therefore, we now provide free service of sharing components in Venus distributed mining pool, and welcome more small miners to access.

> We plan to open the quota of 15 miners, which might be adjusted according to the actual operation in the future. Welcome to contact us through these platforms:

- Slack ([fil-venus channel of filecoinproject.slack.com](https://filecoinproject.slack.com/archives/CEHHJNJS3))
- Github (https://github.com/filecoin-project/venus/discussions )
- Wechat (Filecoin-Venus)
- E-mail (venus@ipfsforce.com)

**Thanks to the contributors**

Thanks to the following developers and contributors for their contribution to Venus and Filecoin:
[@frrist](https://github.com/frrist)
[@anorth](https://github.com/anorth)
[@laser](https://github.com/laser)
[@ZenGround0](https://github.com/ZenGround0)
[@dignifiedquire](https://github.com/dignifiedquire)
[@acruikshank](https://github.com/acruikshank)
[@travisperson](https://github.com/travisperson)
[@whyrusleeping](https://github.com/whyrusleeping)
[@gmasgras](https://github.com/gmasgras)
[@phritz](https://github.com/phritz)
[@shannonwells](https://github.com/shannonwells)
[@rosalinekarr](https://github.com/rosalinekarr)
[@hunjixin](https://github.com/hunjixin)
[@ognots](https://github.com/ognots)
[@icorderi](https://github.com/icorderi)
[@mishmosh](https://github.com/mishmosh)
[@sidke](https://github.com/sidke)
[@porcuquine](https://github.com/porcuquine)
[@diwufeiwen](https://github.com/diwufeiwen)
[@simlecode](https://github.com/simlecode)
[@ingar](https://github.com/ingar)
[@hannahhoward](https://github.com/hannahhoward)
[@Stebalien](https://github.com/Stebalien)
[@aboodman](https://github.com/aboodman)
[@gmas](https://github.com/gmas)
[@anoemi](https://github.com/anoemi)
[@Rennbon](https://github.com/Rennbon)
[@deaswang](https://github.com/deaswang)
[@mslipper](https://github.com/mslipper)
[@lsw45](https://github.com/lsw45)
[@Kubuxu](https://github.com/Kubuxu)
[@arajasek](https://github.com/arajasek)
[@waynewyang](https://github.com/waynewyang)
[@magik6k](https://github.com/magik6k)
[@raulk](https://github.com/raulk)
[@eefahy](https://github.com/eefahy)
[@ridewindx](https://github.com/ridewindx)
[@pooja](https://github.com/pooja)
[@schomatis](https://github.com/schomatis)
[@mburns](https://github.com/mburns)
[@contrun](https://github.com/contrun)
[@lanzafame](https://github.com/lanzafame)
[@Coderlane](https://github.com/Coderlane)
[@jscode017](https://github.com/jscode017)
[@olizilla](https://github.com/olizilla)
[@rafael81](https://github.com/rafael81)
[@anacrolix](https://github.com/anacrolix)
[@danigrant](https://github.com/danigrant)
[@qiwaa](https://github.com/qiwaa)
[@iand](https://github.com/iand)
[@michellebrous](https://github.com/michellebrous)
[@dayu26](https://github.com/dayu26)
[@nijynot](https://github.com/nijynot)
[@ergastic](https://github.com/ergastic)
[@hsanjuan](https://github.com/hsanjuan)
[@iRyanBell](https://github.com/iRyanBell)
[@zeroecco](https://github.com/zeroecco)
[@momack2](https://github.com/momack2)
[@Daniel-Wang](https://github.com/Daniel-Wang)
[@alanshaw](https://github.com/alanshaw)
[@steven004](https://github.com/steven004)
[@sternhenri](https://github.com/sternhenri)
[@zl03jsj](https://github.com/zl03jsj)
[@RichardLitt](https://github.com/RichardLitt)
[@jackoelv](https://github.com/jackoelv)
[@firmianavan](https://github.com/firmianavan)
[@ta0li](https://github.com/ta0li)
[@GFZRZK](https://github.com/GFZRZK)
……

**Comment by sharli91** [comment]
hello someone can explaime how install my minner on ubuntu 18.04 LTS.

**Reply by Joss-Hua** [reply]
Do you mean lotus or Venus?

**Comment by cyruscc** [comment]
金星矿池有文档吗？可以尝试连接吗

**Reply by Joss-Hua** [reply]
可以的。

这里是Venus相关的中英文档：https://venus.filecoin.io

这里是面向社区存储提供者的分布式存储池相关项目：https://venus.filecoin.io/zh/incubation/Rules.html
